### PR TITLE
Feature/ui improvements

### PR DESCRIPTION
**Pull request by cmanamela**
Implemented the alerts filter, alerts pagination and reducers and actions for retrieving individual segment and sensor from the backend

### CODE REVIEW

File: waterlog/src/Components/NavComponent.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ class NavComponent extends Component {
 	}
 
 	componentDidMount() {
-		this.props.fetchAlerts();
+		//this.props.fetchAlerts();
```
Comment (FredJEhlers) [inline waterlog/src/Components/NavComponent.js]: GC

File: waterlog/src/Components/SegmentLeak.js [hunk 2]
```diff
@@ -59,25 +58,26 @@ class SegmentLeak extends Component {
...
-		const alert = alerts.filter((alert) => alert.entityId === parseInt(this.props.match.params.id) 
-			&& alert.date == this.props.match.params.date )[0];
-		
 		const leakInfo = (
 			<div>
-				<div className={`leakInfo ${parseInt(alert.status) == 2 ? alert.severity : 'leak-resolved'}`}>
-					<h2>{`${alert.entityName} ${alert.entityId} ${alert.entityType}`}</h2>
-					<p>({alert.severity})</p>
-					<h1>R {alert.cost.toFixed(2)}</h1> 
+				<div className={`leakInfo ${parseInt(selectedSegment.status) == 2 ? selectedSegment.severity : 'leak-resolved'}`}>
```
Comment (FredJEhlers) [inline waterlog/src/Components/SegmentLeak.js]: Split over multiple lines and line 76

File: waterlog/src/Components/SensorDiagram.js [hunk 3]
```diff
@@ -4,34 +4,35 @@ import { getSensorLayout } from './../utils';
...
         const faultySensor = this.props.sensorId;
-        const sensorSetup = getSensorLayout(parseInt(faultySensor));
+		const sensorSetup = getSensorLayout(parseInt(faultySensor));
+		const resolvedStatus = this.props.status;
 		return (
 			<div id="sensor-diagram">
 				<div id="line" />
 				<div id="first" className="sensor-wrapper">
-					<div className={`sensor ${parseInt(faultySensor) === sensorSetup[0] ? "faulty" : ""}`}>
+					<div className={`sensor ${parseInt(faultySensor) === sensorSetup[0] ? (resolvedStatus == 1 ? "resolved-sensor": "faulty") : ""}`}>
```
Comment (FredJEhlers) [inline waterlog/src/Components/SensorDiagram.js]: Split over multiple lines and lines 16 22 25 32 35

File: waterlog/src/Pages/Alert.js [hunk 4]
```diff
@@ -3,11 +3,26 @@ import AlertTable from "../Components/AlertTable";
...
+      filterOpen: !this.state.filterOpen
+    })
+  }
+
   render() {
     return (
         <div >
-          <AlertTable />
-          <AlertsFilter />
+          { this.state.filterOpen ? <AlertsFilter close={this.toggleFilter} /> : <AlertTable openFilter={this.toggleFilter}/> }
```
Comment (FredJEhlers) [inline waterlog/src/Pages/Alert.js]: Split over multiple lines

File: waterlog/src/Stylesheets/_alerts.scss [hunk 5]
```diff
@@ -68,5 +72,5 @@ td:first-child {
 .alerts-header {
   align-content: center;
   font-weight: normal;
-
-}
+  padding-top: 25px;
+}
```
Comment (FredJEhlers) [inline waterlog/src/Stylesheets/_alerts.scss]: Check for empty lines at EOF

File: waterlog/src/Stylesheets/_alertsFilter.scss2 [hunk 6]
```diff
@@ -0,0 +1,53 @@
+.filter-contents {
```
Comment (FredJEhlers) [inline waterlog/src/Stylesheets/_alertsFilter.scss2]: I think this was not supposed to be commited

File: waterlog/src/Stylesheets/_sensor.scss [hunk 7]
```diff
@@ -65,13 +65,24 @@ small {
...
-    background-color: #FF1744;
+    background-color: $blue;
     border: 10px solid #dd1717;
     color: #f6f6f6;
 }
 
+.resolved-sensor {
+    background-color: $turquoise-blue;
+    border: 10px solid rgba(86,204,247,0.64);
+    color: #f6f6f6;
```
Comment (FredJEhlers) [inline waterlog/src/Stylesheets/_sensor.scss:76]: Choose RGB or hex and implement it across the project

File: waterlog/src/actions/Types.js [hunk 8]
```diff
@@ -114,6 +114,22 @@ export const FETCH_TANK_GRAPH_BEGIN = 'FETCH_TANK_GRAPH_BEGIN';
 export const FETCH_TANK_GRAPH_SUCCESS = 'FETCH_TANK_GRAPH_SUCCESS';
 export const FETCH_TANK_GRAPH_FAILURE='FETCH_TANK_GRAPH_FAILURE';
 
+export const FETCH_FILTERED_ALERTS_BEGIN = 'FETCH_FILTERED_ALERTS_BEGIN';
+export const	FETCH_FILTERED_ALERTS_SUCCESS = 'FETCH_FILTERED_ALERTS_SUCCESS';
```
Comment (FredJEhlers) [inline waterlog/src/actions/Types.js]: indentation

**Review by FredJEhlers — CHANGES_REQUESTED** [review]

File: waterlog/src/Components/NavComponent.js (hunk 1 above)
Comment (cmanamela) [reply waterlog/src/Components/NavComponent.js]: sorted

File: waterlog/src/Components/SensorDiagram.js (hunk 3 above)
Comment (cmanamela) [reply waterlog/src/Components/SensorDiagram.js]: Sorted

File: waterlog/src/Components/SegmentLeak.js (hunk 2 above)
Comment (cmanamela) [reply waterlog/src/Components/SegmentLeak.js]: Sorted

File: waterlog/src/Pages/Alert.js (hunk 4 above)
Comment (cmanamela) [reply waterlog/src/Pages/Alert.js]: sorted

File: waterlog/src/Stylesheets/_alerts.scss (hunk 5 above)
Comment (cmanamela) [reply waterlog/src/Stylesheets/_alerts.scss]: Sorted

File: waterlog/src/Stylesheets/_alertsFilter.scss2 (hunk 6 above)
Comment (cmanamela) [reply waterlog/src/Stylesheets/_alertsFilter.scss2]: Removed

File: waterlog/src/actions/Types.js (hunk 8 above)
Comment (cmanamela) [reply waterlog/src/actions/Types.js]: Sorted

**Review by FredJEhlers — APPROVED** [review]

**Pull request merged**